shared/constants/chat.js

```diff
@@ -177,6 +195,31 @@ export type UpdateLatestMessage = NoErrorTypedAction<'chat:updateLatestMessage',
 export type UpdateMetadata = NoErrorTypedAction<'chat:updateMetadata', {users: Array<string>}>
 export type UpdatedMetadata = NoErrorTypedAction<'chat:updatedMetadata', {[key: string]: MetaData}>
 
+export type EditMessage = NoErrorTypedAction<'chat:editMessage', {message: Message}>
+export type DeleteMessage = NoErrorTypedAction<'chat:deleteMessage', {message: Message}>
+export type ClickedAttach = NoErrorTypedAction<'chat:clickedAttach', {conversationIDKey: ConversationIDKey, filename: string, title: string}>
+export type UploadProgress = NoErrorTypedAction<'chat:uploadProgress', {
+  bytesComplete: number,
+  bytesTotal: number,
+}>
+export type DownloadProgress = NoErrorTypedAction<'chat:downloadProgress', {
+  bytesComplete: number,
+  bytesTotal: number,
+  conversationIDKey: ConversationIDKey,
+}>
+export type LoadAttachment = NoErrorTypedAction<'chat:loadAttachment', {
+  messageID: MessageID,
+  conversationIDKey: ConversationIDKey,
+  loadPreview: boolean,
+  filename: string,
+}>
+export type AttachmentLoaded = NoErrorTypedAction<'chat:attachmentLoaded', {
```

Comment:
This means that you have an attachment and you've requested data for it. Either the preview image or you are in the process of downloaded it. When you get this action either you have the preview or the full file depending on the `isPreview` flag